kernel/libs/aster-util/src/id_allocator.rs

```diff
@@ -63,6 +63,78 @@ impl IdAlloc {
         }
     }
 
+    /// Allocates a consecutive range of new `id`s.
+    ///
+    /// The `count` is the number of consecutive `id`s to allocate. If it is 0, return `None`.
+    ///
+    /// If allocation is not possible, it returns `None`.
+    pub fn alloc_range(&mut self, count: usize) -> Option<Range<usize>> {
+        if count == 0 {
+            return None;
+        }
+
+        let mut allocated_range: Option<Range<usize>> = None;
+        for id in self.first_available_id..self.bitset.len() {
+            // Not enough space left to allocate the range
+            let allocated_count = allocated_range.as_ref().map_or(0, |range| range.len());
+            if count - allocated_count > self.bitset.len() - id {
+                return None;
+            }
+
+            // Reset the range
+            if self.is_allocated(id) {
+                allocated_range = None;
+                continue;
+            }
+
+            match allocated_range {
+                Some(ref mut range) => {
+                    range.end += 1;
+                }
+                None => {
+                    allocated_range = Some(id..id + 1);
+                }
+            }
+
+            let range = allocated_range.as_ref().unwrap();
```

Comment:
When the range starting from first_available_id does not meet the requirements, it needs to be reset, with the start of the new range being determined only in a subsequent loop. In this case, using an Option seems more convenient for implementation.

Reply:
I propose the following revised version. Note how the core logic that determines `allocated_range` is simplified from the original version (reducing the original big loop to only 7 lines and eliminating the branches caused by `Option`). Also, the entire function is broken down into three sections, making it the main workflow more obvious than the original version.

```rust
impl IdAlloc {
    pub fn alloc_consecutive(&mut self, count: usize) -> Option<Range<usize>> {
        if count == 0 {
            return None;
        }

        // Scan the bitmap from the position `first_available_id` 
        // for the first `count` number of consecutive 0's.
        let allocated_range = {
            // Invariance: all bits within `curr_range` are 0's
            let mut curr_range = self.first_available_id..self.first_available_id + 1;
            while curr_range.len() < count && curr_range.end < self.bitmap.len() {
                if !self.is_allocated(curr_range.end) {
                   curr_range.end += 1;
                } else {
                   curr_range = (curr_range.end + 1, curr_range.end + 1);
                }
            }

            if curr_range.len() < count {
                return None;
            }

            curr_range
        };

        // Set every bit to 1 within the allocated range
        for id in &alloc_range {
            self.bitset.set(id, true);
        }

        // In case we need to update first_available_id
        if self.is_allocated(self.first_available_id) {
            self.first_available_id = (range.end..self.bitset.len())
                .find(|&i| !self.bitset[i])
                .map_or(self.bitset.len(), |i| i);
        }

        Some(allocated_range)
    }
}
```

I don't have the time to compile and test the code. So there could be typos or bugs.